File: c/cs50/unit1/CASH/cash.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    int change = 0;

    //Makes sure that a positive value is given
    while (change <= 0)
    {
        change = (get_float("Change owed: ")  * 100+.5);
    }

    int coins = 0;

    //Cycle through the possible options
    while (change > 0)
    {
        if (change >= quarter)
        {
            change -= quarter;
            coins++;
        }
        else if (change >= dime)
        {
            change -= dime;
            coins++;
        }
        else if (change >= nickel)
        {
            change -= nickel;
            coins++;
        }
        else if (change >= penny)
        {
            change -= penny;
            coins++;
        }
    }

    printf("%d\n",coins);
}

File: cs50/cs50/unit6/sentimental/cash/cash.py
# Returns in cents
def get_change(changestr):
    # Checks for decimal number
    if "." not in changestr:
        return int(changestr.strip())*100
    else:
        # Split and add
        change = 0
        splt = changestr.split(".")
        change += int(splt[0])*100

        if int(splt[1][0]) == 0:
            change += int(splt[1][1])
        else:
            if len(splt[1]) == 2:
                change += int(splt[1])
            else:
                change += int(splt[1])*10

        return change


# Define values
quarter, dime, nickel, penny = 25, 10, 5, 1
change = ""
while change == "":
    # Make sure that value is assigned.
    try:
        userin = input("Change owed: ")
        if "-" not in userin:  # Check for negative number
            change = get_change(userin)
    except ValueError:
        print("Change must be a valid number!")

coins = 0

# Get coins
while change > 0:
    if change >= quarter:
        change -= quarter
        coins += 1
    elif change >= dime:
        change -= dime
        coins += 1
    elif change >= nickel:
        change -= nickel
        coins += 1
    else:
        change -= penny
        coins += 1

print(coins)